core/conversion/converters/impl/cast.cpp

```diff
@@ -0,0 +1,51 @@
+#include <torch/torch.h>
+#include "core/conversion/converters/converters.h"
+#include "core/util/prelude.h"
+#include "core/util/trt_util.h"
+
+namespace trtorch {
+namespace core {
+namespace conversion {
+namespace converters {
+namespace impl {
+namespace {
+
+bool register_cast_layer(ConversionCtx* ctx, const torch::jit::Node* n, nvinfer1::ITensor* input, int output_dtype) {
+  auto aten_to_trt_dtype_map = util::get_aten_trt_type_map();
+  TRTORCH_CHECK(
+      aten_to_trt_dtype_map.find(static_cast<at::ScalarType>(output_dtype)) != aten_to_trt_dtype_map.end(),
+      "Conversion to desired datatype is not supported");
+
+  auto cast_layer = ctx->net->addIdentity(*input);
+  cast_layer->setName(util::node_info(n).c_str());
+  cast_layer->setOutputType(0, aten_to_trt_dtype_map.at(static_cast<at::ScalarType>(output_dtype)));
+
+  auto output = ctx->AssociateValueAndTensor(n->outputs()[0], cast_layer->getOutput(0));
+  LOG_DEBUG("Cast layer output tensor shape: " << output->getDimensions());
+
+  return true;
+}
+
+auto cast_registrations TRTORCH_UNUSED =
+    RegisterNodeConversionPatterns()
+        .pattern(
+            {"aten::to.dtype(Tensor self, int dtype, bool non_blocking=False, bool copy=False, int? memory_format=None) -> (Tensor)",
+             [](ConversionCtx* ctx, const torch::jit::Node* n, args& args) -> bool {
+               auto self = args[0].ITensorOrFreeze(ctx);
+               auto output_dtype = args[1].unwrapToScalar().to<int>();
+               return register_cast_layer(ctx, n, self, output_dtype);
+             }})
+        .pattern(
+            {"aten::to.device(Tensor self, Device device, int dtype, bool non_blocking=False, bool copy=False, int? memory_format=None) -> (Tensor)",

```

Comment:
Can we look into the semantics here? it seems like device and dtype are to overloads of to that look at different arguments. Do we need to support the device overload as a converter?

Reply:
We could probably lower out device to calls

Reply:
So there are two variants of torch.to usage
1. `torch.to(torch.float32)`
2. `torch.to(torch.device('cuda:0', torch.float32)`
[https://pytorch.org/docs/stable/tensors.html#torch.Tensor.to](url)
Are you suggesting lowering of device overload ?

Reply:
Okay. for some reason your second comment didn't show up when I saw. I'll move it to lowering